algorithms/integration/__init__.py

```diff
@@ -53,3 +53,15 @@ def __str__(self):
             if value
         ]
         return tabulate(rows)
+
+    def __add__(self, other):
+        assert isinstance(other, TimingInfo)
```

Comment:
suggest something along the lines of
```suggestion
        if not isinstance(other, TimingInfo):
            raise TypeError(f"unsupported addition: expected type TimingInfo, not {other!r}")
```